lib/commands/general.js

```diff
@@ -47,6 +47,19 @@ commands.getWindowSize = async function () {
   return await this.uiautomator2.jwproxy.command('/window/current/size', 'GET', {});
 };
 
+commands.execute = async function (script, args) {
+  if (script.match(/^mobile\:/)) {
+    script = script.replace(/^mobile\:/, '').trim();
+    if (script === 'scroll' && !this.isWebContext()) {
+      let elementId = args[0].element;
+      return await this.uiautomator2.jwproxy.command(`/appium/element/${elementId}/scroll_to`, 'POST', {});
+    } else if (script === 'viewportScreenshot' && !this.isWebContext()) {
+      return await this.getViewportScreenshot();
+    }
+  }
+  this.executeCommand(script, args);
```

Comment:
I don't think this is what should be done at this point. This is a completely new function for this driver, so there is no other behaviour that should be accounted for. 

`executeCommand` is the central command executor for the system, putting commands into the queue and handling everything about command execution. Calling it here will almost certainly break things in ways that would be difficult to fix.

See https://github.com/appium/appium-base-driver/blob/master/lib/basedriver/driver.js#L195-L270

Reply:
We do need to handle the web context, though.